setuptools_rust/build.py

```diff
@@ -162,12 +167,19 @@ def build_extension(self, ext):
                 args.extend(
                     ["-C", "link-arg=-undefined", "-C", "link-arg=dynamic_lookup"]
                 )
+            # Tell musl targets not to statically link libc. See
+            # https://github.com/rust-lang/rust/issues/59302 for details.
+            if b'target_env="musl"' in rust_target_info:
+                rustflags += " -C target-feature=-crt-static"
 
         if not quiet:
             print(" ".join(args), file=sys.stderr)
 
         if ext.native:
-            env["RUSTFLAGS"] = "-C target-cpu=native"
+            rustflags += " -C target-cpu=native"
```

Comment:
I know this was an env var already, though I think it should similarly be `args.extend(["-C", "target-cpu=native"])`?

Reply:
Not sure. I was trying to avoid changing things unnescarily.

Reply:
Based on my comments above, I think this has to be RUSTFLAGS, since the goal is to build _all_ libraries at this optimization level.